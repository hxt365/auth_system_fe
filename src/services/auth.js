// @flow

import {
  LOGIN,
  LOGOUT,
  REFRESH,
  RESET_PASSWORD,
  CHANGE_PASSWORD,
  SIGNUP,
  CONFIRM_EMAIL,
} from 'constants/api';

import type {
  loginFormType,
  resetPasswordType,
  changePasswordFormType,
  signupFormType,
} from 'type';

import customAxios from './axios';

const login = async (data: loginFormType) => {
  const res = await customAxios.post(LOGIN, data).catch(err => err);
  return res;
};

const logout = async () => {
  const res = await customAxios.post(LOGOUT).catch(err => err);
  return res;
};

const resetPassword = async (data: resetPasswordType) => {
  const res = await customAxios.post(RESET_PASSWORD, data).catch(err => err);
  return res;
};

const refresh = async () => {
  const res = await customAxios.post(REFRESH).catch(err => err);
  return res;
};

const changePassword = async (data: changePasswordFormType) => {
  const res = await customAxios.post(CHANGE_PASSWORD, data).catch(err => err);
  return res;
};

const signup = async (data: signupFormType) => {
  const res = await customAxios.post(SIGNUP, data).catch(err => err);
  return res;
};

const confirmEmail = async (token: String) => {
  const link = `${CONFIRM_EMAIL + token}/`;
  const res = await customAxios.post(link).catch(err => err);
  return res;
};

export default {
  login,
  logout,
  refresh,
  resetPassword,
  changePassword,
  signup,
  confirmEmail,
};
