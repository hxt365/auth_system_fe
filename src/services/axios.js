// @flow

import axios from 'axios';

export const BASE_API_URL: String = `${process.env.REACT_APP_BASE_API_URL}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
  withCredentials: true,
  xsrfCookieName: 'csrftoken',
  xsrfHeaderName: 'X-CSRFToken',
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    console.log(error.response);
    return Promise.reject(error);
  },
);

export default customAxios;
