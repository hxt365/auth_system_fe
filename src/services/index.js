export { default as authServices } from './auth';
