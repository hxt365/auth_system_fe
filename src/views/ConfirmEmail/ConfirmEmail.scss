.confirm-email {
  padding-top: 5rem;
}
