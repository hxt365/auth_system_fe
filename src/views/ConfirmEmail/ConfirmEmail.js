// @flow

import React, { useState, useEffect } from 'react';
import { Result, Button, Icon } from 'antd';
import { withRouter, Link } from 'react-router-dom';
import './ConfirmEmail.scss';
import { LOGIN } from 'constants/route';
import { authServices } from 'services';
import Spinner from 'components/share/Spinner';

function ConfirmEmail({ match }: { match: any }) {
  const [confimed, setConfirmed] = useState(false);
  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    (async function IFE() {
      const { token } = match.params;
      const res = await authServices.confirmEmail(token);
      setLoaded(true);
      if (res.status === 200) setConfirmed(true);
      else setConfirmed(false);
    })();
  }, [match.params]);

  let view;
  if (!loaded) view = <Spinner />;
  else
    view = confimed ? (
      <Result
        icon={<Icon type="safety-certificate" theme="twoTone" />}
        title="Congrats! We're glad you joined us!"
        extra={(
          <Button type="primary">
            <Link to={LOGIN}>Log in</Link>
          </Button>
)}
      />
    ) : (
      <Result
        icon={<Icon type="bug" theme="twoTone" />}
        title="The token is not correct or expired!"
      />
    );

  return <section className="confirm-email">{view}</section>;
}

export default withRouter(ConfirmEmail);
