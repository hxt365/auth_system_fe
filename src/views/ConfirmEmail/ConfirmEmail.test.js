import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { authServices } from 'services';
import { BrowserRouter } from 'react-router-dom';
import ConfirmEmail from './ConfirmEmail';

describe('test ConfirmEmail', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<ConfirmEmail />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('make sure confirmEmail called once', () => {
    const confirmEmail = jest.spyOn(authServices, 'confirmEmail').mockImplementation(() => {});
    const wrapper = mount(
      <BrowserRouter>
        <ConfirmEmail />
      </BrowserRouter>,
    );
    expect(confirmEmail).toBeCalledTimes(1);
    confirmEmail.mockClear();
    wrapper.unmount();
  });
});
