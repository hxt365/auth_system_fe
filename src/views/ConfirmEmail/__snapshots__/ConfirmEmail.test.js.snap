// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ConfirmEmail test snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
